src/main/java/io/cdap/plugin/google/common/GoogleDriveFilteringClient.java

```diff
@@ -72,7 +72,16 @@ public List<File> getFilesSummary(List<ExportedType> exportedTypes, int filesNum
       int retrievedFiles = 0;
       int actualFilesNumber = filesNumber;
       if (IdentifierType.FILE_IDENTIFIER.equals(config.getIdentifierType())) {
-        files.add(service.files().get(config.getFileIdentifier()).setSupportsAllDrives(true).execute());
+        File file = service.files().get(config.getFileIdentifier()).setSupportsAllDrives(true).execute();
+        // size will be 1 in case of Sheets plugin with export type as SPREADSHEETS.
+        if (exportedTypes.size() == 1 && exportedTypes.contains(ExportedType.SPREADSHEETS) && !file.getMimeType()
```

Comment:
quick question, will `exportedTypes` in case of `Drive plugin` will always be `> 1`?

Reply:
not always, sometimes it will have but it will read files as binary files with offset and body as schema.

Reply:
I think this is becoming hacky now, can you please create a new class `GoogleSheetsFilteringClient` which extends this class and override this method?

Reply:
done